### ISSUE TITLE
Could not find com.jme3:jme3-core:3.0.10.

### ISSUE THREAD

**Issue by etorth** [methusalah/OpenRTS#106]
Hi,

I'm trying to run this excellent open source software but seems the
./gradlew run

won't work, it reports following errors
I am really new to all these things, could you please have a look?

Best,
Anhong

Configuration on demand is an incubating feature.
:core:compileJava

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all dependencies for configuration ':core:compile'.
> Could not find com.jme3:jme3-core:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-core/3.0.10/jme3-core-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-effects:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-effects/3.0.10/jme3-effects-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-networking:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-networking/3.0.10/jme3-networking-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-blender:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-blender/3.0.10/jme3-blender-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-jbullet:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-jbullet/3.0.10/jme3-jbullet-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-lwjgl:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-lwjgl/3.0.10/jme3-lwjgl-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1
> Could not find com.jme3:jme3-niftygui:3.0.10.
  Searched in the following locations:
      file:/home/anhong/.m2/repository/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      file:/home/anhong/.m2/repository/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
      https://repo1.maven.org/maven2/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      https://repo1.maven.org/maven2/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      https://oss.sonatype.org/content/repositories/snapshots/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      https://oss.sonatype.org/content/repositories/releases/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      http://updates.jmonkeyengine.org/maven/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
      http://jcenter.bintray.com/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.pom
      http://jcenter.bintray.com/com/jme3/jme3-niftygui/3.0.10/jme3-niftygui-3.0.10.jar
  Required by:
      OpenRTS:core:0.0.1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 17.634 secs

**Comment by Pesegato** [comment]
I'm not sure but maybe https://github.com/brainless-studios/OpenRTS is more mantained...

**Comment by methusalah** [comment]
The JME repository has moved and the gradle dependency must be updated.
The new repository is located at org.jmonkeyengine.

You may look at https://github.com/methusalah/alchemist/blob/master/build.gradle and copy/paste the lines. I'm not sur that the 3.0 version of jmonkey is available. I'm sorry but I can't do that myself right now.

**Comment by Protroid** [comment]
I am sorry if I am merely misunderstanding, but what lines am I supposed to copy? Should I replace the contents of build.gradle with the one you linked, or only replace some specific lines?

**Comment by methusalah** [comment]
fixed with https://github.com/methusalah/OpenRTS/commit/ef277c00f4367f0339fb454796058eb55ca988e2

compiles but not tested

**Comment by Protroid** [comment]
I'm still having some issues. Is there perhaps another step I need to accomplish before running OpenRTS for the first time?

C:\Users\Beau\Desktop\OpenRTS>gradlew.bat run
Configuration on demand is an incubating feature.
:core:compileJava

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all dependencies for configuration ':core:compile'.
> Could not find any version that matches com.jme3:jme3-core:[3.0.10,).

I can provide the full text if necessary, but it is highly similar to what etorth posted.

Is there perhaps a pre-requisite for using OpenRTS that I didn't consider? I have never used JMonkey before, and I want to learn so I can use your tool.

**Comment by mrlem** [comment]
If I may: I was able to compile using (in build.xml):

In the repos
`		maven { url "http://dl.bintray.com/jmonkeyengine/com.jme3" }`

In the deps
```
		compile "com.jme3:jme3-core:$jmonkeyengine_version"
		compile "com.jme3:jme3-effects:$jmonkeyengine_version" 
		compile "com.jme3:jme3-networking:$jmonkeyengine_version"
		compile "com.jme3:jme3-blender:$jmonkeyengine_version"
		compile "com.jme3:jme3-bullet:$jmonkeyengine_version"
		compile "com.jme3:jme3-bullet-natives:$jmonkeyengine_version"
		compile "com.jme3:jme3-lwjgl:$jmonkeyengine_version"
		compile "com.jme3:jme3-niftygui:$jmonkeyengine_version"
```

And changing a few Vector2f.interpolateLocal calls to vector2f.interpolate (interpolateLocal couldn't be found). After this, there are some runtime errors (flyCam being null).

All in all, I suspect there were some API changes in JME (though I'm not an expert).

**Comment by Dularl** [comment]
I don't know if this is related but I get:

```
INFO: Audio Renderer Information
 * Device: OpenAL Soft
 * Vendor: OpenAL Community
 * Renderer: OpenAL Soft
 * Version: 1.1 ALSOFT 1.15.1
 * Supported channels: 64
 * ALC extensions: ALC_ENUMERATE_ALL_EXT ALC_ENUMERATION_EXT ALC_EXT_CAPTURE ALC_EXT_DEDICATED ALC_EXT_disconnect ALC_EXT_EFX ALC_EXT_thread_local_context ALC_SOFT_loopback
 * AL extensions: AL_EXT_ALAW AL_EXT_DOUBLE AL_EXT_EXPONENT_DISTANCE AL_EXT_FLOAT32 AL_EXT_IMA4 AL_EXT_LINEAR_DISTANCE AL_EXT_MCFORMATS AL_EXT_MULAW AL_EXT_MULAW_MCFORMATS AL_EXT_OFFSET AL_EXT_source_distance_model AL_LOKI_quadriphonic AL_SOFT_buffer_samples AL_SOFT_buffer_sub_data AL_SOFTX_deferred_updates AL_SOFT_direct_channels AL_SOFT_loop_points AL_SOFT_source_latency
Jan 30, 2017 9:03:44 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
WARNING: Pausing audio device not supported.
Jan 30, 2017 9:03:44 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio effect extension version: 1.0
Jan 30, 2017 9:03:44 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio max auxiliary sends: 4
Jan 30, 2017 9:03:44 PM com.jme3.app.LegacyApplication handleError
SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
java.lang.NullPointerException
        at app.MainRTS.simpleInitApp(MainRTS.java:46)
        at com.jme3.app.SimpleApplication.initialize(SimpleApplication.java:220)
        at app.OpenRTSApplication.initialize(OpenRTSApplication.java:133)
        at com.jme3.system.lwjgl.LwjglAbstractDisplay.initInThread(LwjglAbstractDisplay.java:130)
        at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:211)
        at java.lang.Thread.run(Thread.java:745)

> Building 33% > :core:run
```

**Comment by methusalah** [comment]
> All in all, I suspect there were some API changes in JME (though I'm not an expert).

Yes it has been some changes with backward compatibility issues. I should have used a jMonkey release instead of a working branch but jMonkey was new to github at the time.

I've tried the current gradle config on a fresh install and it works. I suspect some env issues on your side.
please check that you have imported the three projects in the same workspace, have refresh gradle hard, have have followed [the wiki](https://github.com/methusalah/OpenRTS/wiki/Set-up-the-project-with-your-favorite-IDE-(Eclipse,-Netbeans,-etc.))

**Comment by Protroid** [comment]
If it is anything to note, the suggestion by mrlem did help the code compile, but now I am getting a null pointer exception at Thread[jME3 Main,5,main] when the program attempts to launch JMonkey.

I'll report back here / in a new issue if I can figure out anything about that.

**Comment by RamakrishnanArun** [comment]
@mrlem 's post above solved the dependency issues. I seem to have some compilation issues and bad symbols in the source files but that's another issue.

**Comment by RamakrishnanArun** [comment]
@Protroid were you able to find a way around that problem? I'm just trying it out today and I have the same error.

**Comment by RamakrishnanArun** [comment]
I see what causes it.

`OpenRTSApplication.initialize()` calls `super.initialize()` (which is a Monkey class). That in turn calls the `simpleInitApp` method down in `MainRTS`. The `flyCam` variable is referenced here. However it is initialized in `OpenRTSApplication.initialize()` only after the call to `super.initialize()`.

I don't know enough at this point to change anything though.

**Comment by Protroid** [comment]
@RamakrishnanArun No, I was not able to get around the Null Pointer Exception, college is a joy :P

**Comment by RamakrishnanArun** [comment]
@methusalah can you help? I'm interested and trying to set this up.

**Comment by dosaki** [comment]
I decided to try this on Windows after I got #107 on Linux.

I'm having the same issue as @RamakrishnanArun and @Protroid now.

**Comment by lorduriel** [comment]
After that ju just need to get rid of 

`com.jme3.app.LegacyApplication `

in app.OpenRTSApplication.java

and change LegacyApplication  to Application in the class
